common/persistence/visibility/visibility_manager_impl.go

```diff
@@ -295,6 +296,21 @@ func (p *visibilityManagerImpl) newInternalVisibilityRequestBase(
 		return nil, err
 	}
 
+	var searchAttrs *commonpb.SearchAttributes
+	if len(request.SearchAttributes.GetIndexedFields()) > 0 {
```

Comment:
This change seems to modify the behavior of how search attributes are handled and does more than just suppress the error as the PR title suggests. The PR title should capture this imo

Why does the new error-suppression make it necessary to remove system Search Attributes from the map here?

Reply:
Because we don't want to write them to database. The query to insert/upsert would basically fail since it would try to specify the system column twice (once by the visibility store and another here).